### DISCUSSION TITLE
setLocale repeat sluggable for all available locales in DB

### DISCUSSION THREAD

**Discussion by mohamedsabil83** [spatie/laravel-sluggable#220] [Q&A]
I'm using sluggable and translatable packages. have two languages Arabic and English. let's say we have a Member model which has two columns (**name** and **slug**) and all settings set properly.

when using `setTranslation` method like:
```php
$member = Member::first();
$member->setTranslation('name', 'ar', 'أحمد');
$member->setTranslation('name', 'en', 'Ahmed');
$member->save();
```
the slug column sets as expected `{"ar":"أحمد","en":"ahmed"}`

but when using `setLocale` method like following, it clones the slug using the last locale:

**The Arabic last**
```php
$member = Member::first();
$member->setLocale('en');
$member->name = "Ahmed";
$member->setLocale('ar');
$member->name = "أحمد";
$member->save();
```
the slug column will be `{"ar":"أحمد","en":"أحمد"}`

**The English last**
```php
$member = Member::first();
$member->setLocale('ar');
$member->name = "أحمد";
$member->setLocale('en');
$member->name = "Ahmed";
$member->save();
```
the slug column will be `{"ar":"ahmed","en":"ahmed"}`

The correct name column must saved as `{"ar":"أحمد","en":"Ahmed"}`

**Comment by mohamedsabil83** [comment]
To be sure, is it only me who faced this issue?

**Reply by mohamedsabil83** [reply]
any hints about that issue? it turns to discussion and nothing after that

**Reply by truffolone** [reply]
Are you in a different context than basic laravel?

I ask this because if you are using backpack with `Backpack\CRUD\app\Models\Traits\SpatieTranslatable\HasTranslations` instead of the default Spatie HasTranslations Trait, the $this->withLocale in HasTranslatableSlug:37 will not work properly using only the default locale

**Reply by mohamedsabil83** [reply]
No, in basic laravel. I try many things but nothing works. So, I have to use `->setTranslation('key', 'locale', 'value');` for the field will be slugged.
